### PR TITLE
[Yaml] deprecate "? " starting unquoted strings

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | yes
| Tests pass?   | yes
| Fixed tickets | #20579
| License       | MIT
| Doc PR        |

### CODE REVIEW

File: UPGRADE-3.3.md [hunk 1]
```diff
@@ -252,8 +252,12 @@ Workflow
 Yaml
 ----
 
- * Deprecated support for implicitly parsing non-string mapping keys as strings. Mapping keys that are no strings will
-   lead to a `ParseException` in Symfony 4.0. Use the `PARSE_KEYS_AS_STRINGS` flag to opt-in for keys to be parsed as
+ * starting an unquoted string with a question mark followed by a space is
```
Comment (javiereguiluz) [inline UPGRADE-3.3.md]: `starting ...` -> `Starting ...` in all UPGRADE files.

File: UPGRADE-4.0.md [hunk 2]
```diff
@@ -463,8 +463,12 @@ Workflow
 Yaml
 ----
 
- * Removed support for implicitly parsing non-string mapping keys as strings. Mapping keys that are no strings will
-   result in a `ParseException`. Use the `PARSE_KEYS_AS_STRINGS` flag to opt-in for keys to be parsed as strings.
+ * starting an unquoted string with a question mark followed by a space is
```
Comment (javiereguiluz) [inline UPGRADE-4.0.md]: `is throws` -> `throws`

**Review by javiereguiluz — APPROVED** [review]
👍

**Review by nicolas-grekas — APPROVED** [review]
:+1:

**Comment by fabpot** [comment]
Thank you @xabbuh.

**Pull request merged**